### ISSUE TITLE
Inverse of `.ignore` file (like `.include`) or explicit include support in `.ignore` file

### ISSUE THREAD

**Issue by etosan** [ggreer/the_silver_searcher#1542]
`ag` and some other tools support `.ignore` file, where one can exclude searching in specific files and directories that are of no interest.

Unfortunately in some projects, like for example `gitea` managed ones, there are hidden dirs like `.gitea` that one would like `ag` to search through anyway, by default, especially when dealing with IaC monorepos and similar (as action workflows live there etc).

It would be great, if one could have explicit inclusions for such dirs defined in `.ignore`-file for exemaple by following `.gitignore`'s `!something` pattern to explicitly include given path.

Would this be possible?

**Comment by thomasjm** [comment]
I was expecting this to work with "!" syntax and was surprised to find that it doesn't.

Seems like `ripgrep` supports it out of the box.

**Comment by etosan** [comment]
@thomasjm I am hitting this problem more and more recently, and it seems that development rate on `ag` has slowed down. Would you consider `ripgrep` sufficient replacement for `ag` feature wise?

Inability to negate exclusions is becoming show-stopper for me.

**Comment by thomasjm** [comment]
Well, `ripgrep` has almost twice the GitHub stars. 

There are a lot of details comparing them in this blog post: https://burntsushi.net/ripgrep/

At this point I only use `ag` because the Emacs Helm integration is better than the `ripgrep` equivalent.

**Comment by etosan** [comment]
Yeah with `ag<->ripgrep` theme I see myself using `ripgrep` more and more, locally, just because of missing this feature. It seems that `ag` has entered it's "maintenance only" mode project stage. Only reason why I still use `ag` are remote enterprise distros where `rg` will take some time to settle.
